### PR TITLE
feat(batch): ComputeEnvironment implements IConnectable

### LINKED ISSUE #20983: (aws-batch): ComputeEnvironment should implement IConnectable
### Describe the feature

Once a ComputeEnvironment is created, there is no way to access the security group associated with it. We would like to implement IConectable interface in the ComputeEnvironment to more easily control connections from the batch container.

### Use Case

Current implementation requires 3 steps below when creating batch with db connection.
1. create a security group
2. create the computing resource and associate with the security group created in 1.
3. run `db.connections.allowFrom(sg, ...)`

If ComputeEnvironment implements IConnectable, it become more simple
1. create computing environment without explicit security group association
2. run `db.connections.allowFrom(computeEnv, ...)`

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.24

### Environment details (OS name and version, etc.)

ubuntu 18.04

### PR DESCRIPTION
**Pull request by tcutts**
closes #20983

ComputeEnvironments have embedded security groups in them.  These are currently difficult to reach and modify in CDK stacks, which forces users into undesirable practices when integrating batch queues with other services such as as EFS filesystems or RDS instances.  Ideally, it should be possible to use compute environments as a target:

```ts
something.connections.allowDefaultPortFrom(computeEnvironment);
```

but this isn't currently possible, so the user may try to work around it by allowing from any IPv4, or by having to use an escape hatch, which is not simple.

This pull request adds ec2.IConnectable to batch.ComputeEnvironment.  It still seems to pass all the existing tests and integration tests, so I don't think it's a breaking change, but I suspect it could be done in a better way than I've done it, to make things even simpler for the user.

----

### All Submissions:

* [yes] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [no] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [yes] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [yes] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mrgrain** [comment]
Hi @tcutts and thanks for the contribution! This looks exciting and I think you've found the right approach to implement this. Before I go ahead with a more thorough review, can you please make sure the build passes and with this kind of logic change, we will definitely need unit tests in addition to the integration test you've already added.

**Comment by tcutts** [comment]
There you go - builds properly now.  My example in the README needed fixing...

**Review by TheRealAmazonKendra — DISMISSED** [review]
Looks like we still need unit tests on this change.

**Review by mrgrain — CHANGES_REQUESTED** [review]
A few minor changes and since we are changing the logical, I'd like to see a unit test.

File: packages/@aws-cdk/aws-batch/test/integ.batch-with-efs.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      SECRET: ecs.Secret.fromSecretsManager(secret),
+    },
+    mountPoints,
+    volumes,
+  },
+});
+
+new integ.IntegTest(app, 'BatchWithEFSTest', {
+  testCases: [stack],
+});
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-batch/test/integ.batch-with-efs.ts:201]:
```suggestion
});

app.synth();
```

**Comment by tcutts** [comment]
I've added a couple of unit tests, but it's not a particular area of expertise for me (not being an SDE!); I've borrowed some of the connections test logic from aws-ec2, and re-purposed it here to check that the right ingress rules are created, for both fargate and EC2 managed compute environments.  Is that sufficient?

**Review by mrgrain — APPROVED** [review]
Yes, these look great! Thank you for your work here. :shipit:

**Pull request merged**